src/server.c

```diff
@@ -6958,6 +6950,11 @@ int main(int argc, char **argv) {
         serverLog(LL_WARNING,"Server initialized");
     #ifdef __linux__
         linuxMemoryWarnings();
+        sds err_msg;
+        if (check_xen(&err_msg) < 0) {
+            serverLog(LL_WARNING, "WARNING %s", err_msg);
```

Comment:
You sure? The check now has no disadvantages, and in all known cases we don't want `xen` clocksource. Lets add this only in the future if needed.

Reply:
i thought someone may be bothered by the warning print.
but on a second thought, we don't have any way to disable the THP or other warnings either..
so ok.. but @yossigo please ack.

Reply:
I wonder if we can move these check codes into syscheck.c altogether.

Reply:
@sundb what do you mean? (not sure of the context of your last message)

Reply:
@oranagra Now all the checking methods have been moved to syscheck.c, but there is still some checking code in server.c, I think it is more reasonable to put them all together.

Reply:
ok, but we did that already (except for the tcp_backlog one that's tied to a config value).
and this comment of yours is in a thread about whether or not to add a `checkIgnoreWarning` to silence the print.
so i'm not certain about the context, maybe i'm missing your intention.

Reply:
@sundb the logic is that `--check-system` is used to run **all** available system checks and set an appropriate exit code. This can be done by install scripts, etc. to verify the system is correctly configured. This means the checks can also be inefficient or even leave the process in a problematic state. On the other hand the checks which are executed on normal server startup have to be fast and not affect the process's state.

@yossigo I changed the warning messages based on your comment. 

I'm resolving this. Re-open if needed.

Reply:
@yoav-steinberg Ohh, I missed it, I thought it was just optimization.
@oranagra If so, I don't think it seems appropriate to brink the #10467 check here, when building a docker image, this check will not fail, because it is not built under qemu, until the image is actually running in qemu, it will be will fail.